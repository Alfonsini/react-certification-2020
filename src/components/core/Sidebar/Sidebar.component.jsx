import React, { useState } from 'react';

import { Link } from 'react-router-dom';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faPlay } from '@fortawesome/free-solid-svg-icons';

import MenuVertical from '../Menus/Vertical.component';

import './Sidebar.styles.css';

function Sidebar() {
  const [hover, setHover] = useState(false);

  const toggleHover = () => {
    setHover(!hover);
  };

  return (
    <aside
      onMouseOver={() => toggleHover()}
      onMouseOut={() => toggleHover()}
      onFocus={() => {}}
      onBlur={() => {}}
      className={hover ? 'navbar-aside-container expanded' : 'navbar-aside-container'}
    >
      <ul className="navbar-aside-menu header-brand">
        <li>
          <Link to="/" className="nav-links">
            <FontAwesomeIcon icon={faPlay} />
            <span className="navbar-aside-logo text-truncate">Player</span>
          </Link>
        </li>
      </ul>
      <div className="shadow-bottom" />
      <MenuVertical />
    </aside>
  );
}

export default Sidebar;
