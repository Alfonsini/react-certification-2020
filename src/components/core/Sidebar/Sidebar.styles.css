.navbar-aside-container {
  z-index: 1031;
  --sidebar-width: 4.2em;
  height: 100vh;
  left: 0;
  width: var(--sidebar-width);
  background-color: #020635;
  position: fixed;
  transition: width 0.2s ease;
}

.navbar-aside-menu {
  font-size: 1.1rem;
  font-family: 'Montserrat', Helvetica, Arial, serif;
  font-weight: 400;
  overflow-y: hidden;
  overflow-x: hidden;
  list-style: none;
  margin: 0;
  padding: 0;
  padding-bottom: 0.75rem;
}

.navbar-aside-menu.header-brand {
  margin-top: 1rem;
}

.navbar-aside-menu li {
  position: relative;
  white-space: nowrap;
  margin-bottom: 18px;
}

.navbar-aside-menu li.active {
  z-index: 1;
}

.navbar-aside-menu li a {
  color: #6d6f97;
  margin: 0 15px;
  padding: 10px 15px 10px 15px;
  line-height: 1.45;
  text-overflow: ellipsis;
  overflow: hidden;
  align-items: center;
  width: calc(260px - 35px);
  display: flex;
}

.navbar-aside-container:not(.expanded) li a {
  width: 50px;
}

.navbar-aside-container ul.navbar-aside-menu:not(.header-brand) li:hover a > * {
  transition: transform 0.25s;
  transform: translateX(5px);
}

.navbar-aside-menu li a svg {
  margin-right: 1.1rem;
}

.navbar-aside-menu li.active a {
  color: #dcdcdc;
  background: linear-gradient(118deg, #7367f0, #7367f0b3);
  box-shadow: 0 0 10px 1px rgb(115 103 240 / 70%);
  border-radius: 4px;
}

.navbar-aside-container:not(.expanded) li.active a {
  background: #292f78;
  box-shadow: none;
}

ul.navbar-aside-menu:not(.header-brand) li:first-child {
  margin-top: 0.5rem;
}

.navbar-aside-menu .navbar-aside-logo {
  color: #7367f0;
}

.navbar-aside-menu.header-brand li a {
  color: #7367f0;
}

.navbar-aside-container.expanded {
  width: 260px;
}

.navbar-aside-container .shadow-bottom {
  display: none;
  position: absolute;
  z-index: 2;
  height: 50px;
  width: 100%;
  pointer-events: none;
  filter: blur(5px);
}

.text-truncate {
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

@media screen and (min-width: 768px) {
  .navbar-aside-container {
    display: table-cell;
  }
}

@media screen and (max-width: 500px) {
  .navbar-aside-container {
    display: none;
  }
}
