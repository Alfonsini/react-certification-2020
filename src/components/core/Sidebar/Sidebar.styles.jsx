import styled from 'styled-components';

const StyledAside = styled.aside`
  z-index: 1031;
  height: 100vh;
  left: 0;
  width: ${(props) => (props.expanded ? '260px' : '4.2em')};
  background-color: #020635;
  position: fixed;
  transition: width 0.2s ease;

  @media screen and (min-width: 768px) {
    display: table-cell;
  }

  @media screen and (max-width: 500px) {
    display: none;
  }
`;

const StyledShadowBottom = styled.div`
  display: none;
  position: absolute;
  z-index: 2;
  height: 50px;
  width: 100%;
  pointer-events: none;
  filter: blur(5px);
`;

export { StyledAside, StyledShadowBottom };
