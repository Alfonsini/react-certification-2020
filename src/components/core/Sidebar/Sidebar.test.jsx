import React from 'react';
import { render, screen, fireEvent } from '../../../test/testUtils';

import Sidebar from '.';

beforeEach(() => {
  render(<Sidebar />);
});

describe('Sidebar', () => {
  test('sidebar render correctly', async () => {
    const aside = screen.getByRole('menubar');
    expect(aside).toMatchSnapshot();
  });

  test('sidebar render correctly not expanded', async () => {
    const aside = screen.getByRole('menubar');
    fireEvent.mouseOver(aside);
    fireEvent.mouseOut(aside);

    const style = window.getComputedStyle(aside);
    expect(style).toMatchSnapshot();
  });

  test('sidebar render correctly expanded', async () => {
    const aside = screen.getByRole('menubar');
    fireEvent.mouseOver(aside);

    const style = window.getComputedStyle(aside);
    expect(style).toMatchSnapshot();
  });
});
