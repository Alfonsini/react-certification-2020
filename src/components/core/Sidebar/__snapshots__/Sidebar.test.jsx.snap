// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar sidebar render correctly 1`] = `
<aside
  class="sc-citwmv jXTUgu"
  role="menubar"
>
  <ul
    class="sc-iktFzd iPqmEP"
    role="list"
  >
    <li
      class="sc-jJEJSO kLQdpR"
    >
      <a
        class="sc-hiSbYr elvejx"
        to="/"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-play fa-w-14 "
          data-icon="play"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
            fill="currentColor"
          />
        </svg>
        <span
          class="sc-cBNfnY bpVrrY"
        >
          Player
        </span>
      </a>
    </li>
  </ul>
  <div
    class="sc-jcVebW jqwWBN"
  />
  <ul
    class="sc-iJuUWI DfGXm"
    role="list"
  >
    <li
      class="sc-giIncl dVdkAt"
    >
      <a
        class="sc-ezrdKe eoejhV"
        expanded="0"
        href="/"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-home fa-w-18 "
          data-icon="home"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
            fill="currentColor"
          />
        </svg>
        <span
          class="sc-kLgntA inqwmJ"
        >
          Home
        </span>
      </a>
    </li>
  </ul>
</aside>
`;

exports[`Sidebar sidebar render correctly expanded 1`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "box-sizing",
  "10": "visibility",
  "2": "z-index",
  "3": "height",
  "4": "left",
  "5": "width",
  "6": "background-color",
  "7": "box-shadow",
  "8": "position",
  "9": "transition",
  "_importants": Object {
    "background-color": "",
    "box-shadow": "",
    "box-sizing": "",
    "display": "",
    "height": "",
    "left": "",
    "position": "",
    "transition": "",
    "visibility": undefined,
    "width": "",
    "z-index": "",
  },
  "_length": 11,
  "_onChange": [Function],
  "_values": Object {
    "background-color": "rgb(2, 6, 53)",
    "box-shadow": "0 0 15px 0 rgb(34 41 47 / 5%)",
    "box-sizing": "inherit",
    "display": "block",
    "height": "100vh",
    "left": "0px",
    "position": "fixed",
    "transition": "width 0.2s ease",
    "visibility": "visible",
    "width": "260px",
    "z-index": "1031",
  },
}
`;

exports[`Sidebar sidebar render correctly not expanded 1`] = `
CSSStyleDeclaration {
  "0": "display",
  "1": "box-sizing",
  "10": "visibility",
  "2": "z-index",
  "3": "height",
  "4": "left",
  "5": "width",
  "6": "background-color",
  "7": "box-shadow",
  "8": "position",
  "9": "transition",
  "_importants": Object {
    "background-color": "",
    "box-shadow": "",
    "box-sizing": "",
    "display": "",
    "height": "",
    "left": "",
    "position": "",
    "transition": "",
    "visibility": undefined,
    "width": "",
    "z-index": "",
  },
  "_length": 11,
  "_onChange": [Function],
  "_values": Object {
    "background-color": "rgb(2, 6, 53)",
    "box-shadow": "0 0 15px 0 rgb(34 41 47 / 5%)",
    "box-sizing": "inherit",
    "display": "block",
    "height": "100vh",
    "left": "0px",
    "position": "fixed",
    "transition": "width 0.2s ease",
    "visibility": "visible",
    "width": "4.2em",
    "z-index": "1031",
  },
}
`;
